scripts/exregional_process_radarref.sh

```diff
@@ -0,0 +1,348 @@
+# forecast.  These are needed below for various operations.
+#
+#-----------------------------------------------------------------------
+#
+#START_DATE=$(echo "${CDATE}" | sed 's/\([[:digit:]]\{2\}\)$/ \1/')
+START_DATE=$(echo "${PDY} ${cyc}")
+YYYYMMDDHH=$(date +%Y%m%d%H -d "${START_DATE}")
+#JJJ=$(date +%j -d "${START_DATE}")
+
+YYYY=${YYYYMMDDHH:0:4}
+MM=${YYYYMMDDHH:4:2}
+DD=${YYYYMMDDHH:6:2}
+HH=${YYYYMMDDHH:8:2}
+#YYYYMMDD=${YYYYMMDDHH:0:8}
+
+#
+#-----------------------------------------------------------------------
+#
+# Find cycle type: cold or warm 
+#  BKTYPE=0: warm start
+#  BKTYPE=1: cold start
+#
+#-----------------------------------------------------------------------
+#
+BKTYPE=0
+if [ ${DO_SPINUP} == "TRUE" ]; then
+  if [ ${CYCLE_TYPE} == "spinup" ]; then
+    for cyc_start in "${CYCL_HRS_SPINSTART[@]}"; do
+      if [ ${HH} -eq ${cyc_start} ]; then
+        BKTYPE=1
+      fi
+    done
+  fi
+else
+  for cyc_start in "${CYCL_HRS_PRODSTART[@]}"; do
+    if [ ${HH} -eq ${cyc_start} ]; then
+        BKTYPE=1
+    fi
+  done
+fi
+
+n_iolayouty=$(($IO_LAYOUT_Y-1))
+list_iolayout=$(seq 0 $n_iolayouty)
+
+#
+#-----------------------------------------------------------------------
+#
+# Loop through different time levels
+# Get into working directory
+#
+#-----------------------------------------------------------------------
+#
+print_info_msg "$VERBOSE" "
+Getting into working directory for radar reflectivity process ..."
+
+for bigmin in ${RADARREFL_TIMELEVEL[@]}; do
+  bigmin=$( printf %2.2i $bigmin )
+  mkdir_vrfy ${DATA}/${bigmin}
+  cd ${DATA}/${bigmin}
+
+  pregen_grid_dir=$DOMAIN_PREGEN_BASEDIR/${PREDEF_GRID_NAME}
+
+  print_info_msg "$VERBOSE" "pregen_grid_dir is $pregen_grid_dir"
+
+#
+#-----------------------------------------------------------------------
+#
+# link or copy background files
+#
+#-----------------------------------------------------------------------
+
+  if [ ${BKTYPE} -eq 1 ]; then
+    cp_vrfy ${pregen_grid_dir}/fv3_grid_spec          fv3sar_grid_spec.nc
+  else
+    if [ "${IO_LAYOUT_Y}" == "1" ]; then
+      cp_vrfy ${pregen_grid_dir}/fv3_grid_spec          fv3sar_grid_spec.nc
+    else
+      for ii in $list_iolayout
+      do
+        iii=$(printf %4.4i $ii)
+        cp_vrfy ${pregen_grid_dir}/fv3_grid_spec.${iii}   fv3sar_grid_spec.nc.${iii}
+      done
+    fi
+  fi
+
+#
+#-----------------------------------------------------------------------
+#
+# link/copy observation files to working directory 
+#
+#-----------------------------------------------------------------------
+
+
+case $MACHINE in
+
+"WCOSS2")
+
+  obs_appendix=grib2.gz
+  ;;
+"JET" | "HERA" | "ORION")
+
+  obs_appendix=grib2
+esac
+
+  NSSL=${OBSPATH_NSSLMOSIAC}
+
+  mrms="MergedReflectivityQC"
+
+# Link to the MRMS operational data
+  echo "bigmin = ${bigmin}"
+  echo "RADARREFL_MINS = ${RADARREFL_MINS[@]}"
+
+# Link to the MRMS operational data
+  for min in ${RADARREFL_MINS[@]}
```

Comment:
Maybe add a comment here that this internal loop is looking for files within close proximity to the time level in question?